libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/DecryptStream.swift

```diff
@@ -95,6 +95,11 @@ public class DecryptStream: InputStream {
             do {
                 let data = Data(bytes: encryptedBuffer, count: bytesRead)
                 let decryptedData = try Encryptor.decrypt(data: data, using: key)
+                guard decryptedData.count <= CryptStream.chunkSize else {
```

Comment:
SFS upgraded to Xcode 13.3.x and started seeing file decryption issues only in their release builds. 

It seems like something weird is happening here where we know that the decrypted data that’s being added to the buffer is bound by our encryption block size but the compiler doesn’t and is handling it differently somehow unless we have an explicit check for it

`precondition(decryptedData.count <= CryptStream.chunkSize)` also works instead but I didn’t want to be that aggressive even if it should never trigger

I have a bigger change that does more refactoring, but opened this on `master` so they could pick up the least intrusive fix now